darwin/client.py

```diff
@@ -17,21 +20,43 @@
     Unauthorized,
 )
 from darwin.utils import is_project_dir, urljoin
-from darwin.validators import name_taken, validation_error
 
 
 class Client:
-    def __init__(self, config: Config, default_team: Optional[str] = None):
+    def __init__(self, config: Config, log: Logger, default_team: Optional[str] = None):
```

Comment:
We should have a default logger defined somewhere, e.g. in `logger.py`

Reply:
And then we would import it using `from darwin.logger import log` or something simillar, right?
I also understand we can test this mocking the logger like we do with requests. 
Plus we would not change the Client's API even further. 

I like the idea, @andreaazzini what do you think?

Reply:
I agree with your suggestions, however the PR is already big enough and this change falls outside of the scope.
This PR's main objective was to introduce a rather simple usage of `logging` which it already does.

I created a Linear Ticket to work on this on the future:
https://linear.app/v7labs/issue/V7-2469/have-a-centralized-point-for-logger-configuration